pkg/controllers/provisioning/scheduling/suite_test.go

```diff
@@ -2053,25 +2064,37 @@ var _ = Describe("Instance Type Selection", func() {
 		cloudProv.CreateCalls = nil
 		cloudProv.InstanceTypes = nil
 		minPrice = math.MaxFloat64
-		// create 100+ instance types with varying CPU/memory/architecture
+		// Create 1000+ unique instance types with varying CPU/memory/architecture/OS/zone/capacity type.
 		instanceTypePrices = map[string]float64{}
 		for _, cpu := range []int{1, 2, 4, 8, 16, 32, 64} {
-			for mem := range []int{1, 2, 4, 8, 16, 32, 64, 128} {
-				for _, arch := range []string{v1alpha5.ArchitectureAmd64, v1alpha5.ArchitectureArm64} {
-					it := fake.NewInstanceType(fake.InstanceTypeOptions{
-						Name:             fmt.Sprintf("%d-cpu-%d-mem-%s", cpu, mem, arch),
-						Architecture:     arch,
-						OperatingSystems: sets.NewString("linux", "windows", "darwin"),
-						Resources: v1.ResourceList{
-							v1.ResourceCPU:    resource.MustParse(fmt.Sprintf("%d", cpu)),
-							v1.ResourceMemory: resource.MustParse(fmt.Sprintf("%dGi", mem)),
-						},
-					})
-					instanceTypePrices[it.Name()] = it.Price()
-					if it.Price() < minPrice {
-						minPrice = it.Price()
+			for _, mem := range []int{1, 2, 4, 8, 16, 32, 64, 128} {
```

Comment:
These suites are getting pretty big. Thoughts on separating them out into multiple logical suites?

Reply:
Ginkgo only lets you have one suite per package as far as I can tell.  I can at least separate it into multiple files though.